admin/class-smaily-for-wp-admin.php

```diff
@@ -120,7 +120,11 @@ public function smaily_subscription_block_init( $screen ) {
 			true
 		);
 
-		wp_add_inline_script( $this->plugin_name, 'var autoresponders = ' . json_encode( $autoresponders ) . ';' );
+		$json_autoresponders = wp_json_encode( $autoresponders );
+		$json_autoresponders = str_replace( '"', '\"', $json_autoresponders );
+		$json_autoresponders = str_replace( "'", "\'", $json_autoresponders );
+
+		wp_add_inline_script( $this->plugin_name, sprintf( "var autoresponders = '%s';", $json_autoresponders ), 'before' );
```

Comment:
A more solid solution would be to wrap the JSON object with [`JSON.stringify`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify), which would handle all of the necessary escaping:

```suggestion
		wp_add_inline_script( $this->plugin_name, sprintf( "var autoresponders = JSON.stringify(" . wp_json_encode( $autoresponders ) . ");", 'before' );
```

Reply:
Switched to JSON.stringify https://github.com/sendsmaily/sendsmaily-wordpress-plugin/pull/142/commits/f3297f44afc9251c2571f5b070bc0eb393697537